test/files/run/t5565.scala

```diff
@@ -1,7 +1,9 @@
+import scala.annotation.nowarn
 import scala.language.reflectiveCalls
 import scala.language.implicitConversions
 
 object Test extends App {
+  @nowarn  // the inferred type includes the default arg, which can't be written explicitly
```

Comment:
@dwijnand inviting your thoughts on your example: implicits should have explicit types, but it is willing to infer a type that can't be written explicitly. In this case, the default arg is not allowed in the refinement. (One could introduce a trait, but that would negate the test.) Probably you will say: "Rules are made to be broken."

Reply:
Nice find!